## Capturing errors in FastAPI

We’ll build a basic FastAPI app. To start, ensure your virtual environment is still active then install Flask:

```bash
pip install fastapi
```

Next, create an `main.py` file with the following basic setup:

```python
from fastapi import FastAPI
from fastapi.responses import JSONResponse

# Initialize FastAPI app
app = FastAPI()

@app.get('/')
async def root():
    return {'message': 'Hello World'}

@app.get('/error')
def trigger_error():
    raise ValueError('This is a triggered backend error!')

@app.exception_handler(Exception)
async def http_exception_handler(request, exc):
    posthog.capture_exception(exc)
    return JSONResponse(status_code=500, content={'message': str(exc)})
```

You can then run `fastapi dev main.py` and go to `http://localhost:8000/error` to see your intentional error (that’s handled by FastAPI).

### Setting up PostHog

import PythonFastAPIExceptionAutocapture from './python-fastapi-exception-autocapture.mdx'

<PythonFastAPIExceptionAutocapture />

When you go to `http://localhost:5000/error` now, you’ll see the error captured in PostHog.